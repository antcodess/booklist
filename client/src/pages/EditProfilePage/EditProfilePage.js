import React from 'react';
import { withRouter } from 'react-router-dom';

import BreadCrumbNav from '../../components/BreadCrumbNav/BreadCrumbNav';
import UpdateProfile from '../../components/UpdateProfile/UpdateProfile';

import './EditProfilePage.scss';

const EditProfilePage = ({ history }) => {
  return(
    <React.Fragment>
      <BreadCrumbNav title='Edit Profile'/>
      <div className='edit-profile'>
        <h2>Edit Profile</h2>
        <span 
          className='back-span' 
          onClick={() => history.push('/account')}>Back to Account</span>
        <UpdateProfile />
      </div>
    </React.Fragment>
  );
}

export default withRouter(EditProfilePage);