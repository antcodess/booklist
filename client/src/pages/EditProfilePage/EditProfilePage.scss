.edit-profile {
  text-align: left;
  h2 {
    margin: 20px 0;
    font-size: 2rem; 
  }

  span.back-span {
    cursor: pointer;

    &:hover {
      opacity: 0.7;
    }
  }
}