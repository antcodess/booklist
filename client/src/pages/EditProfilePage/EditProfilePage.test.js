import React from 'react';
import { shallow } from 'enzyme';
import { EditProfilePage } from './EditProfilePage';

describe('EditProfilePage Component', () => {
  let wrapper;
  let mockHistory;
  
  beforeEach(() => {
    
    mockHistory = { push: jest.fn() };

    const mockProps = {
      history: mockHistory, 
    };
    
    wrapper = shallow(<EditProfilePage { ...mockProps } />);
  });

  it('should render EditProfilePage Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push when clicked for .back-span span', () => {
    wrapper.find('.back-span').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });
});