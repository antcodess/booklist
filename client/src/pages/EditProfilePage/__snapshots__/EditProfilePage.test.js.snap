// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfilePage Component should render EditProfilePage Component 1`] = `ShallowWrapper {}`;
